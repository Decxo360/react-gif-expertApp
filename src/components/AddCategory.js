import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({handleAdd}) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputChange = ({target}) => {
    setInputValue(target.value);
  }
  
  const handleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return; 
    // setCategory(category => [inputValue,...category]);
    setInputValue("")
    handleAdd(inputValue.trim())
  }

  return (
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={inputValue}
          onChange={handleInputChange}
        />
      </form>
  )
}

// AddCategory.propTypes = {
//   setCategory: PropTypes.func.isRequired
// }