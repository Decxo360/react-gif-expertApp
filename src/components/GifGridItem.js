import React from 'react'
import PropTypes from "prop-types"

export const GifGridItem = ({title,url}) => {
  return (
    <div className='card animate__animated animate__bounce'>
        <img src={url} alt={title}></img>
        <h3>{title}</h3>
    </div>
  )
}
GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}