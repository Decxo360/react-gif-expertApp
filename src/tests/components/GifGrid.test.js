import React from 'react';
import '@testing-library/jest-dom'
import { GifGrid } from "../../components/GifGrid"
import { shallow } from 'enzyme'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Trabajando en el componente GifGrid.js', () => {

    
    test('Debe de mostrarse correctamente', () => { 

        useFetchGifs.mockReturnValue({
            data:[],
            loading:true,
        });
        const wrapper= shallow(<GifGrid category="samurai x"/>);
        expect(wrapper).toMatchSnapshot();

     });
    
     test('Debe de mostrar items cuando se cargan imagenes', () => { 

        const gifs = [{
            id:'123a',
            url:'https://yuutu.com',
            title:'cualquiercosa'
        }];

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });
        const wrapper= shallow(<GifGrid category="samurai x"/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
      });
})