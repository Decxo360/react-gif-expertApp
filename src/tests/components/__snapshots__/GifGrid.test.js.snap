// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trabajando en el componente GifGrid.js Debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3>
    samurai x
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123a"
      key="123a"
      title="cualquiercosa"
      url="https://yuutu.com"
    />
  </div>
</Fragment>
`;

exports[`Trabajando en el componente GifGrid.js Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    samurai x
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
